src/security/decorators.py

```diff
@@ -1164,3 +1164,39 @@ def keyword_page_enabled_wrapper(request, *args, **kwargs):
             return func(request, *args, **kwargs)
 
     return keyword_page_enabled_wrapper
+
+
+def submission_authorised(func):
+    """
+    Checks if roles are required to access submission page.
+    :param func:
+    :return:
+    """
+
+    @base_check_required
+    def submission_authorised_wrapper(request, *args, **kwargs):
+        if request.repository and request.repository.limit_access_to_submission:
+            if not preprint_models.RepositoryRole.objects.filter(
+                repository=request.repository,
+                user=request.user,
+                role__slug='author',
+            ).exists():
+                return redirect(
+                    reverse(
+                        'request_submission_access'
+                    )
+                )
+
+        if request.journal and request.journal.get_setting('general', 'limit_access_to_submission'):
+            if not request.user.is_author(
+                request,
+            ):
+                return redirect(
+                    reverse(
+                        'request_submission_access'
+                    )
+                )
+
```

Comment:
Shoudl editors and stafff/superusers have the ability to submit even if the setting is enabled and they lack the author role?

Reply:
Good catch. `is_author` works for staff but not for editors and obviously works for neither on repos here.

Suggest: `if staff or (repo and is moderator) or (journal and is editor): return func` right at the start before the logic ?

Reply:
ỳeah, there is an `editor_or_manager` decorator in this file we might be able to re-purpose